diff -Naur a/src/libstd/process.rs b/src/libstd/process.rs
--- a/src/libstd/process.rs	2020-04-15 12:39:45.294925564 +0200
+++ b/src/libstd/process.rs	2020-04-15 12:39:59.382863103 +0200
@@ -1741,8 +1741,8 @@
     fn signal_reported_right() {
         use crate::os::unix::process::ExitStatusExt;
 
-        let mut p =
-            Command::new("/bin/sh").arg("-c").arg("read a").stdin(Stdio::piped()).spawn().unwrap();
+        let mut p = Command::new("/bin/sh")
+            .arg("-c").arg("read a").stdin(Stdio::piped()).spawn().unwrap();
         p.kill().unwrap();
         match p.wait().unwrap().signal() {
             Some(9) => {}
